src/client/extension.ts

```diff
@@ -122,6 +124,21 @@ async function activateUnsafe(
     startupDurations.totalActivateTime = startupStopWatch.elapsedTime - startupDurations.startActivateTime;
     activationDeferred.resolve();
 
+    setTimeout(async () => {
+        if (activatedServiceContainer) {
+            const interpreterManager = activatedServiceContainer.get<IInterpreterService>(IInterpreterService);
+            const workspaceService = activatedServiceContainer.get<IWorkspaceService>(IWorkspaceService);
+            const workspaces = workspaceService.workspaceFolders ?? [];
+            try {
+                await interpreterManager.refresh(workspaces.length > 0 ? workspaces[0].uri : undefined);
+            } catch (ex) {
+                traceError('Python Extension: interpreterManager.refresh', ex);
+            }
```

Comment:
I feel this is clearer.
```suggestion
            await interpreterManager
            .refresh(workspaceService.hasWorkspaceFolders ? workspaceService.workspaceFolders![0].uri : undefined)
            .catch((ex) => traceError('Python Extension: interpreterManager.refresh', ex));
```

Reply:
we want to avoid `!` usage in the extension. I don't think we should go back to the old way. There is a refactoring plan to remove all uses of `workspaceService.hasWorkspaceFolders`, and essentially have `workspaceService.workspaceFolders` return empty array if there is nothing.

Reply:
I was just referring to the `try...catch` thing actually.
```suggestion
            await interpreterManager
            .refresh(workspaces.length > 0 ? workspaces[0].uri : undefined)
            .catch((ex) => traceError('Python Extension: interpreterManager.refresh', ex));
```

Reply:
ahh i see